### DISCUSSION TITLE
BLUETOOTH_SCAN permission

### DISCUSSION THREAD

**Discussion by gilgru23** [kenjdavidson/react-native-bluetooth-classic#179] [General]
![WhatsApp Image 2022-04-02 at 18 45 35](https://user-images.githubusercontent.com/48647298/161390726-617a5312-681e-440e-beac-13e628f1e1be.jpeg)
Hey run into this problem above, what can I do to change the permission in order to run the startDiscovery() method? thanks

**Comment by kenjdavidson** [comment]
Add the appropriate permission to your manifest?

https://developer.android.com/guide/topics/connectivity/bluetooth/permissions



Get Outlook for Android<https://aka.ms/ghei36>
________________________________
From: gilgru23 ***@***.***>
Sent: Saturday, April 2, 2022 11:47:58 AM
To: kenjdavidson/react-native-bluetooth-classic ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [kenjdavidson/react-native-bluetooth-classic] BLUETOOTH_SCAN permission (Discussion #179)


[WhatsApp Image 2022-04-02 at 18 45 35]<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F48647298%2F161390726-617a5312-681e-440e-beac-13e628f1e1be.jpeg&data=04%7C01%7C%7C214c78d3ab7c4ad503ee08da14c029c0%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637845112803436387%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=OdYlHpCmqI8pua4HOCABcT4qIy%2FtD81rKt5HNenumIk%3D&reserved=0>
Hey run into this problem above, what can I do to change the permission in order to run the startDiscovery() method? thanks

**Reply by gilgru23** [reply]
Hey I added the required permission < uses-permission android:name="android.permission.BLUETOOTH_SCAN" />" to every AndroidManifest file in the project and it still doens't work, other functions like isBluetoothAvailable and checkBluetoothEnabled are work

**Comment by kenjdavidson** [comment]
Not sure. It shouldn't need to be added to the libraries, since yours should overwrite. The permission stuff is pretty Android standard.

I know to use the camera you need to request permission in the app itself. Maybe that applies to later Android versions.

Google the expo barcode scanner, the example has a way to ask for permissions in the app itself.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by ghost** [comment]
I think you need to add these permissions to your manifest:

```
<!-- Request legacy Bluetooth permissions on older devices. -->
<uses-permission android:name="android.permission.BLUETOOTH" android:maxSdkVersion="30" />
<uses-permission android:name="android.permission.BLUETOOTH_ADMIN" android:maxSdkVersion="30" />

<!-- Needed only if your app looks for Bluetooth devices.
	 If your app doesn't use Bluetooth scan results to derive physical
	 location information, you can strongly assert that your app
	 doesn't derive physical location. -->
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" />

<!-- Needed only if your app makes the device discoverable to Bluetooth
	 devices. -->
<uses-permission android:name="android.permission.BLUETOOTH_ADVERTISE" />

<!-- Needed only if your app communicates with already-paired Bluetooth
	 devices. -->
<uses-permission android:name="android.permission.BLUETOOTH_CONNECT" />

<!-- Needed only if your app uses Bluetooth scan results to derive physical location. -->
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
    
After adding them to my project everything works fine for me.

**Reply by gilgru23** [reply]
The suggestion of @kenjdavidson  solved my problem, I asked for a permission with AndroidPermission and the functions works well
